packages/core/src/hydration/event_replay.ts

```diff
@@ -79,3 +118,78 @@ export function setJSActionAttribute(
     }
   }
 }
+
+function setEventReplayer(dispatcher: Dispatcher) {
+  dispatcher.setEventReplayer(queue => {
+    for (const event of queue) {
+      handleEvent(event);
+    }
+  });
+}
+
+/**
+ * TODO: Update documentation here.
+ *
+ * @param target Component, Directive or DOM Node.
+ */
+export function getLViewByElement(target: any): LView|null {
+  let lView = readLViewData(target);
+  if (lView) {
+    return lView;
+  } else {
+    // if the context is not found then we need to traverse upwards up the DOM
+    // to find the nearest element that has already been monkey patched with data
+    let parent = target as any;
```

Comment:
nit: we can probably cast to `HTMLElement`:
```suggestion
    let parent = target as HTMLElement;
```

Reply:
done